src/Dropdown/Dropdown.jsx

```diff
@@ -171,6 +177,76 @@ const Dropdown = ({
     [disabled, error, isCollapsed]
   )
 
+  useLayoutEffect(() => {
+    const handleKeyDownEvent = e => {
+      if (isCollapsed) {
+        return
+      }
+
+      if (e.code === 'Escape') {
+        e.preventDefault()
+
+        setCollapsed(true)
+      }
+
+      if (e.code === 'ArrowDown') {
+        e.preventDefault()
+
+        setCurrentKeyboardFocus(hover =>
+          Math.min(options.length - 1, Math.max(-1, hover + 1))
+        )
+      }
+
+      if (e.code === 'ArrowUp') {
+        e.preventDefault()
+
+        setCurrentKeyboardFocus(hover =>
+          Math.min(options.length - 1, Math.max(-1, hover - 1))
+        )
+      }
+
+      if (e.code === 'Enter') {
+        e.preventDefault()
+
+        if (
+          currentKeyboardFocus !== -1 &&
+          !options[currentKeyboardFocus].props.disabled
+        ) {
+          onSelectItem(options[currentKeyboardFocus].props.value)
+        }
+      }
+    }
+
+    const removeCurrentHover = () => {
+      setCurrentKeyboardFocus(-1)
+    }
+
+    const element = wrapper.current
+    element.addEventListener('keydown', handleKeyDownEvent)

```

Comment:
If you have a reference, you can use the `onKeyDown` prop on the target
`onKeyDown={handleKeyDownEvent}`

Reply:
8c6f23348e959e8dd428f3991cfb3f5652b572e7